src/coord/src/persistcfg.rs

```diff
@@ -304,6 +304,24 @@ impl PersisterWithConfig {
                     envelope_details: crate::catalog::SerializedEnvelopePersistDetails::Upsert,
                 })
             }
+            SourceConnector::External {
+                connector: ExternalSourceConnector::Kafka(_),
+                envelope: SourceEnvelope::None(_),
+                ..
+            } if self.config.kafka_upsert_source_enabled => {
```

Comment:
The name of the variable no longer reflects what it does. It should be `--persistent-kafka-sources` or something along those lines, to match `--persistent-kafka-tables`

Reply:
This guy... :sweat_smile: I wanted to bring this up in the channel, yes! I also have a change for DEBEZIUM prepared but wanted to wait until they're all in.

I wanted to suggest just `--persisten-kafka-sources`, but it would mean (for now) that when you enable this it also depends on the ENVELOPE that you're using, which can be a bit unpredictable.

Reply:
> persisten

:D

Reply:
> I wanted to suggest just --persisten-kafka-sources, but it would mean (for now) that when you enable this it also depends on the ENVELOPE that you're using, which can be a bit unpredictable.

I'd be okay with erroring if you turn on the flag and then create a kafka source we don't yet support in persist. The flag is definitely marked experimental so in my mind as long as we're not silently doing something confusing, then we're free to just not support stuff